drivers/volume/csi/csi.go

```diff
@@ -1343,6 +1343,15 @@ func (c *csi) restoreVolumeSnapshot(
 		vsObj.Spec.Source.PersistentVolumeClaimName = nil
 		vsObj.Spec.Source.VolumeSnapshotContentName = &vsc.(*kSnapshotv1.VolumeSnapshotContent).Name
 		vsObj.Namespace = namespace
+		// PB-8482 Add restoreSize to annotations
+		// Convert RestoreSize to string and store it in restoreSizeStr
+		restoreSizeStr := vsObj.Status.RestoreSize.String()
+		// Initialize Annotations map if it is nil
+		if vsObj.Annotations == nil {
+			vsObj.Annotations = make(map[string]string)
+		}
+		// Add the restoreSize annotation to the VolumeSnapshot object
+		vsObj.Annotations["restoreSize"] = restoreSizeStr
```

Comment:
Since this is non px-backup k8s object, can you please prefix px-backup like px-backup/restoreSize